code/game/objects/structures/guillotine.dm

```diff
@@ -0,0 +1,234 @@
+#define GUILLOTINE_DECAP_MIN_SHARP  7  // Minimum amount of sharpness for decapitation. Any less and it will just do severe brute damage
+#define GUILLOTINE_ANIMATION_LENGTH 9 // How many deciseconds the animation is 
+#define GUILLOTINE_BLADE_RAISED     1
+#define GUILLOTINE_BLADE_MOVING     2
+#define GUILLOTINE_BLADE_DROPPED    3
+#define GUILLOTINE_BLADE_SHARPENING 4
+#define GUILLOTINE_HEAD_OFFSET      16 // How much we need to move the player to center their head
+#define GUILLOTINE_LAYER_DIFF       1.2 // How much to increase/decrease a head when it's buckled/unbuckled
+#define GUILLOTINE_ACTIVATE_DELAY   30 // Delay for activating
+
+/obj/structure/guillotine
+	name = "guillotine"
+	desc = "A large structure used to remove the heads of traitors and treasonists."
+	icon = 'icons/obj/guillotine.dmi'
+	icon_state = "guillotine_raised"
+	can_buckle = TRUE
+	anchored = TRUE
+	density = TRUE
+	max_buckled_mobs = 1
+	buckle_lying = FALSE
+	buckle_prevents_pull = TRUE
+	layer = ABOVE_MOB_LAYER
+	var/blade_status = GUILLOTINE_BLADE_RAISED
+	var/blade_sharpness = GUILLOTINE_BLADE_MAX_SHARP // How sharp the blade is
+	var/kill_count = 0
+
+/obj/structure/guillotine/Initialize()
+	LAZYINITLIST(buckled_mobs)
+	. = ..()
+
+/obj/structure/guillotine/examine(mob/user)
+	..()
+
+	var/msg = ""
+
+	msg += "It is [anchored ? "wrenched to the floor." : "unsecured. A wrench should fix that."]<br/>"
+
+	if (blade_status == GUILLOTINE_BLADE_RAISED)
+		msg += "The blade is raised, ready to fall, and"
+
+		if (blade_sharpness >= GUILLOTINE_DECAP_MIN_SHARP)
+			msg += " looks sharp enough to decapitate without any resistance."
+		else
+			msg += " doesn't look particularly sharp. Perhaps a whetstone can be used to sharpen it."
+	else
+		msg += "The blade is hidden inside the stocks."
+
+	if (LAZYLEN(buckled_mobs))
+		msg += "<br/>"
+		msg += "Someone appears to be strapped in. You can help them out, or you can harm them by activating the guillotine."
+
+	to_chat(user, msg)
+
+	return msg
+
+/obj/structure/guillotine/attack_hand(mob/user)
+	add_fingerprint(user)
+
+	switch (blade_status)
+		if (GUILLOTINE_BLADE_MOVING)
+			return
+		if (GUILLOTINE_BLADE_DROPPED)
+			blade_status = GUILLOTINE_BLADE_MOVING
+			icon_state = "guillotine_raise"
+			addtimer(CALLBACK(src, .proc/raise_blade), GUILLOTINE_ANIMATION_LENGTH)
+			return
+		if (GUILLOTINE_BLADE_RAISED)
+			if (LAZYLEN(buckled_mobs))
+				if (user.a_intent == INTENT_HARM)
+					user.visible_message("<span class='warning'>[user] begins to pull the lever!</span>",
+						                 "<span class='warning'>You begin to the pull the lever.</span>")
+					if (do_after(user, GUILLOTINE_ACTIVATE_DELAY, target = src) && blade_status == GUILLOTINE_BLADE_RAISED)
+						blade_status = GUILLOTINE_BLADE_MOVING
+						icon_state = "guillotine_drop"
+						addtimer(CALLBACK(src, .proc/drop_blade), GUILLOTINE_ANIMATION_LENGTH - 2) // Minus two so we play the sound and decap faster
```

Comment:
you -would- need it here however, because this is where the blade falls on a target